Makefile

```diff
@@ -133,126 +136,122 @@ assets: $(GOBINDATA)
 	@./scripts/build-check-comments.sh
 
-# format the code:
-# - format code (including imports format)
-# - clean up all white noise
 .PHONY: format
+format: ## Formats Go code including imports and cleans up white noise.
 format: $(GOIMPORTS) check-comments
 	@echo ">> formatting code"
 	@$(GOIMPORTS) -w $(FILES_TO_FMT)
 	@SED_BIN="$(SED)" scripts/cleanup-white-noise.sh $(FILES_TO_FMT)
 
-# proto generates golang files from Thanos proto files.
 .PHONY: proto
+proto: ## Generates Go files from Thanos proto files.
 proto: check-git  $(GOIMPORTS) $(PROTOC)
 	@GOIMPORTS_BIN="$(GOIMPORTS)" PROTOC_BIN="$(PROTOC)" scripts/genproto.sh
 
 .PHONY: promu
 promu: $(PROMU)
 
 .PHONY: tarballs-release
+tarballs-release: ## Build tarballs.
 tarballs-release: $(PROMU)
 	@echo ">> Publishing tarballs"
 	$(PROMU) crossbuild -v tarballs
 	$(PROMU) checksum -v .tarballs
 	$(PROMU) release -v .tarballs
 
-# test runs all Thanos golang tests against each supported version of Prometheus.
 .PHONY: test
+test: ## Runs all Thanos Go unit tests against each supported version of Prometheus. This excludes tests in ./test/e2e.
 test: export GOCACHE= $(TMP_GOPATH)/gocache
 test: export THANOS_TEST_MINIO_PATH= $(MINIO_SERVER)
 test: export THANOS_TEST_PROMETHEUS_VERSIONS= $(PROM_VERSIONS)
 test: export THANOS_TEST_ALERTMANAGER_PATH= $(ALERTMANAGER)
 test: check-git install-deps
 	@echo ">> install thanos GOOPTS=${GOOPTS}"
-	# Thanos binary is required by e2e tests.
-	@go install github.com/thanos-io/thanos/cmd/thanos
-	# Be careful on GOCACHE. Those tests are sometimes using built Thanos/Prometheus binaries directly. Don't cache those.
-	@rm -rf ${GOCACHE}
-	@echo ">> running all tests. Do export THANOS_TEST_OBJSTORE_SKIP=GCS,S3,AZURE,SWIFT,COS,ALIYUNOSS if you want to skip e2e tests against all real store buckets. Current value: ${THANOS_TEST_OBJSTORE_SKIP}"
-	@go test $(shell go list ./... | grep -v /vendor/);
+	@echo ">> running unit tests (without /test/e2e). Do export THANOS_TEST_OBJSTORE_SKIP=GCS,S3,AZURE,SWIFT,COS,ALIYUNOSS if you want to skip e2e tests against all real store buckets. Current value: ${THANOS_TEST_OBJSTORE_SKIP}"
+	@go test $(shell go list ./... | grep -v /vendor/ | grep -v /test/e2e);
 
 .PHONY: test-ci
+test-ci: ## Runs test for CI, so excluding object storage integrations that we don't have configured yet.
 test-ci: export THANOS_TEST_OBJSTORE_SKIP=AZURE,SWIFT,COS,ALIYUNOSS
 test-ci:
 	@echo ">> Skipping ${THANOS_TEST_OBJSTORE_SKIP} tests"
 	$(MAKE) test
 
 .PHONY: test-local
+test-local: ## Runs test excluding tests for ALL  object storage integrations.
 test-local: export THANOS_TEST_OBJSTORE_SKIP=GCS,S3,AZURE,SWIFT,COS,ALIYUNOSS
 test-local:
 	$(MAKE) test
 
-# install-deps installs dependencies for e2e tetss.
-# It installs supported versions of Prometheus and alertmanager to test against in e2e.
+.PHONY: test-e2e
+test-e2e: ## Runs all Thanos e2e docker-based e2e tests from test/e2e.
+test-e2e: docker
```

Comment:
Can we add a message here before running everything that you either need to have `root` or set up your user in a certain way?

Reply:
done.